test/h2_worker_test.exs

```diff
@@ -439,6 +439,71 @@ defmodule Sparrow.H2WorkerTest do
     end
   end
 
+  test "server correctly starting with successful connection and does not schedule or runs pinging",
+       context do
+    ptest [
+            domain: string(min: 3, max: 10, chars: ?a..?z),
+            port: int(min: 0, max: 65_535),
+            tls_options: list(of: atom(), min: 0, max: 3)
+          ],
+          repeat_for: @repeats do
+      ping_interval = nil
+
+      with_mock H2Adapter,
+        open: fn _, _, _ -> {:ok, context[:connection_ref]} end,
+        ping: fn _ ->
+          :ok
+        end,
+        close: fn _ -> :ok end do
+        config =
+          Config.new(
+            domain,
+            port,
+            context[:auth],
+            tls_options,
+            ping_interval
+          )
+
+        {:ok, pid} = GenServer.start(Sparrow.H2Worker, config)
```

Comment:
How about that we start_link here and don't use Process.exit on line 472?

Reply:
I'll try it :)